### PR TITLE
Migrate the docstring comments when generate connection template

### PR DESCRIPTION
**Pull request by chenslucky**
# Description

1. Migrate the docstring comments from a connection class, using them to provide detailed information about each corresponding key

**Condition 1**: If the connection class doesn't contain any docstring comments, the keys will be listed without any additional information.
``` python
class MyFirstConnection(CustomStrongTypeConnection):
    api_key: Secret
    api_key_2: Secret
    api_base: str
    api_base_2: str
```
``` yaml
name: "to_replace_with_connection_name"
type: custom
custom_type: MyFirstConnection
module: my_tool_package.connections
package: my-tools-package-with-cstc
package_version: 0.0.6
configs:
  api_base: "to_replace_with_api_base"
  api_base_2: "to_replace_with_api_base_2"
secrets:      # must-have
  api_key: "to_replace_with_api_key"
  api_key_2: "to_replace_with_api_key_2"
```
**Condition 2**: If the docstring comments only contain information about specific keys, then these keys will have their corresponding descriptions, while the rest will be listed without any additional information.
``` python
class MyFirstConnection(CustomStrongTypeConnection):
    """Custom connection.

    :param api_key: The api key.
    :type api_key: String
    :param api_base: The api base.
    :type api_base: String
    """
    api_key: Secret
    api_key_2: Secret
    api_base: str
    api_base_2: str
```
``` yaml
name: "to_replace_with_connection_name"
type: custom
custom_type: MyFirstConnection
module: my_tool_package.connections
package: my-tools-package-with-cstc
package_version: 0.0.6
configs:
  api_base: "to_replace_with_api_base"  # String, The api base.
  api_base_2: "to_replace_with_api_base_2"
secrets:      # must-have
  api_key: "to_replace_with_api_key"  # String, The api key.
  api_key_2: "to_replace_with_api_key_2"
```
**Condition 3**: If the docstring comments contain information related to invalid or non-existent keys, this information will be disregarded during the migration process.
``` python
class MyFirstConnection(CustomStrongTypeConnection):
    """Custom connection.

    :param api_key: The api key.
    :type api_key: String
    :param api_base: The api base.
    :type api_base: String
    :param api_key_2: The api key 2.
    :type api_key_2: String
    :param api_base_2: The api base 2.
    :type api_base_2: String
    :param api_key_3: The api key 3.
    :type api_key_3: String
    :param api_base_3: The api base 3.
    :type api_base_3: String
    """
    api_key: Secret
    api_key_2: Secret
    api_base: str
    api_base_2: str
```
``` yaml
name: "to_replace_with_connection_name"
type: custom
custom_type: MyFirstConnection
module: my_tool_package.connections
package: my-tools-package-with-cstc
package_version: 0.0.6
configs:
  api_base: "to_replace_with_api_base"  # String, The api base.
  api_base_2: "to_replace_with_api_base_2" # String, The api base 2.
secrets:      # must-have
  api_key: "to_replace_with_api_key"  # String, The api key.
  api_key_2: "to_replace_with_api_key_2" # String, The api key 2.
```

**Condition 4**: If the docstring comments miss the type or param for one key, only exist comments for this key move to template.
``` python
class MyFirstConnection(CustomStrongTypeConnection):
    """Custom connection.

    :param api_key: The api key.
    :type api_key: String
    :param api_base: The api base.
    """
    api_key: Secret
    api_key_2: Secret
    api_base: str
    api_base_2: str
```
``` yaml
name: "to_replace_with_connection_name"
type: custom
custom_type: MyFirstConnection
module: my_tool_package.connections
package: my-tools-package-with-cstc
package_version: 0.0.6
configs:
  api_base: "to_replace_with_api_base"  # The api base.
  api_base_2: "to_replace_with_api_base_2"
secrets:      # must-have
  api_key: "to_replace_with_api_key"  # String, The api key.
  api_key_2: "to_replace_with_api_key_2"
```

2. Move default value of keys to template
``` python
class MyCustomConnectionWithDefaultValue(CustomStrongTypeConnection):
    api_key: Secret
    api_base: str = "default-api-base"
```
``` yaml
name: "to_replace_with_connection_name"
type: custom
custom_type: MyFirstConnection
module: my_tool_package.connections
package: my-tools-package-with-cstc
package_version: 0.0.6
configs:
  api_base: "default-api-base"
secrets:      # must-have
  api_key: "to_replace_with_api_key"
```

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by chjinche** [comment]
configs:
# :param api_base: The secrets kv pairs. :type api_base: Dict[str, str]
  api_base: <to-be-replaced> # string, the api endpoint.

File: src/promptflow/promptflow/_cli/_pf/_connection.py [hunk 1]
```diff
@@ -138,6 +155,27 @@ def add_connection_list(subparsers):
     )
 
 
+def add_gen_connection_template(subparsers):
+    epilog = """
```
Comment (16oeahr) [inline src/promptflow/promptflow/_cli/_pf/_connection.py:159]: fix spacing

**Comment by 16oeahr** [comment]
> # Description
> This process involves two changes:
> 
> 1. Promptflow Cli support generate custom strong type connection template.
> 
> ```shell
> pf connection gen-template -n my_connection_name -m my_module_name --package my_package_name --output ./output
> ```
> 
> **Tests:** Connection class: image
> 
> Generated yaml file template: image
> 
> 2. Migrate the docstring comments from a connection class, using them to provide detailed information about each corresponding key
> 
> **Condition 1**: If the connection class doesn't contain any docstring comments, the keys will be listed without any additional information.
> 
> ```python
> class MyFirstConnection(CustomStrongTypeConnection):
>     api_key: Secret
>     api_key_2: Secret
>     api_base: str = "This is my first connection."
>     api_base_2: str = "This is my first connection."
> ```
> 
> ```yaml
> name: <connection-name>
> type: custom
> custom_type: MyFirstConnection
> module: my_tool_package.connections
> package: my-tools-package-with-cstc
> package_version: 0.0.6
> configs:
>   api_base: <api-base>
>   api_base_2: <api-base-2>
> secrets:      # must-have
>   api_key: <api-key>
>   api_key_2: <api-key-2>
> ```
> 
> **Condition 2**: If the docstring comments only contain information about specific keys, then these keys will have their corresponding descriptions, while the rest will be listed without any additional information.
> 
> ```python
> class MyFirstConnection(CustomStrongTypeConnection):
>     """Custom connection.
> 
>     :param api_key: The api key.
>     :type api_key: String
>     :param api_base: The api base.
>     :type api_base: String
>     """
>     api_key: Secret
>     api_key_2: Secret
>     api_base: str = "This is my first connection."
>     api_base_2: str = "This is my first connection."
> ```
> 
> ```yaml
> name: <connection-name>
> type: custom
> custom_type: MyFirstConnection
> module: my_tool_package.connections
> package: my-tools-package-with-cstc
> package_version: 0.0.6
> configs:
>   api_base: <api-base>  # String, The api base.
>   api_base_2: <api-base-2>
> secrets:      # must-have
>   api_key: <api-key>  # String, The api key.
>   api_key_2: <api-key-2>
> ```
> 
> **Condition 3**: If the docstring comments contain information related to invalid or non-existent keys, this information will be disregarded during the migration process.
> 
> ```python
> class MyFirstConnection(CustomStrongTypeConnection):
>     """Custom connection.
> 
>     :param api_key: The api key.
>     :type api_key: String
>     :param api_base: The api base.
>     :type api_base: String
>     :param api_key_2: The api key 2.
>     :type api_key_2: String
>     :param api_base_2: The api base 2.
>     :type api_base_2: String
>     :param api_key_3: The api key 3.
>     :type api_key_3: String
>     :param api_base_3: The api base 3.
>     :type api_base_3: String
>     """
>     api_key: Secret
>     api_key_2: Secret
>     api_base: str = "This is my first connection."
>     api_base_2: str = "This is my first connection."
> ```
> 
> ```yaml
> name: <connection-name>
> type: custom
> custom_type: MyFirstConnection
> module: my_tool_package.connections
> package: my-tools-package-with-cstc
> package_version: 0.0.6
> configs:
>   api_base: <api-base>  # String, The api base.
>   api_base_2: <api-base-2> # String, The api base 2.
> secrets:      # must-have
>   api_key: <api-key>  # String, The api key.
>   api_key_2: <api-key-2> # String, The api key 2.
> ```
> 
> **TODO:** Add e2etests and units tests for all change in this pr
> 
> # All Promptflow Contribution checklist:
> * [x]  **The pull request does not introduce [breaking changes]**
> * [ ]  **CHANGELOG is updated for new features, bug fixes or other significant changes.**
> * [x]  **I have read the [contribution guidelines](../CONTRIBUTING.md).**
> 
> ## General Guidelines and Best Practices
> * [x]  Title of the pull request is clear and informative.
> * [x]  There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
> 
> ### Testing Guidelines
> * [ ]  Pull request includes test coverage for the included changes.

1. What would happen if the type field is missing, only param description
2. What would happen if the type and param is mis-written twice?

**Comment by chenslucky** [comment]
Q: What would happen if the type field is missing, only param description
A: only param description

Q: What would happen if the type and param is mis-written twice?
A: join two strings. code logic "key_description = ' '.join(re.findall(param_pattern, doc))"

File: src/promptflow/promptflow/_cli/_pf/_connection.py (hunk 1 above)
Comment (chenslucky) [reply src/promptflow/promptflow/_cli/_pf/_connection.py:159]: fix all spacing in this file

**Comment by chjinche** [comment]
> 1. What would happen if the type field is missing, only param description
> 2. What would happen if the type and param is mis-written twice?
> 3. Currently, the cli command is to support user to create a connection using cli way.
>    image
>    https://microsoft.github.io/promptflow/how-to-guides/manage-connections.html
> 
> ```
> pf connection gen-template -n my_connection_name -m my_module_name --package my_package_name --output ./output
> ```
> 
> @chjinche do you think the command to provide a template is good? Should we rename it to
> 
> ```
> pf connection template create -n my_connection_name -m my_module_name --package my_package_name --output ./output
> ```

`pf connection template create` looks better.

**Comment by chjinche** [comment]
Please add tests covering both happy paths and error handling.

**Review by 16oeahr — APPROVED** [review]

**Review by chjinche — APPROVED** [review]

**Review by dorisjoy — APPROVED** [review]

**Pull request merged**